### PR TITLE
Improve UI responsiveness while enumerating

### PR DESCRIPTION
**Pull request by hez2010**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Contributes to #4180

**Details of Changes**
Add details of changes here.
- Move enumeration to background to unblock UI thread
- Replace lambdas with local functions

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

**Screenshots (optional)**
Before:

https://user-images.githubusercontent.com/14960345/131523833-691f213b-c18e-4aa9-a4e6-888403347fe0.mp4

After:

https://user-images.githubusercontent.com/14960345/131524578-75a96d42-ec29-4ea6-a272-7941f4e4de96.mp4

### CODE REVIEW

**Review by gave92 — DISMISSED** [review]
~Sweet, thank you!~ Edit: I'm sure this is sweet too, but I meant to approve #5992. I'll review this shortly.

File: Files/Filesystem/StorageEnumerators/UniversalStorageEnumerator.cs [hunk 1]
```diff
@@ -186,71 +188,80 @@ CancellationToken cancellationToken
...
+
+                return (icon, iconData, itemThumbnailImgVis, itemEmptyImgVis);
             }
+
             if (cancellationToken.IsCancellationRequested)
             {
                 return null;
             }
 
+            await CoreApplication.MainView.Dispatcher.YieldAsync(Windows.UI.Core.CoreDispatcherPriority.Normal);
```
Comment (gave92) [inline Files/Filesystem/StorageEnumerators/UniversalStorageEnumerator.cs:262]: Could you explain me this one? I understand that now ListEntries is executed on a background thread, what's the point of yielding to the UI thread?

Comment (gave92) [reply Files/Filesystem/StorageEnumerators/UniversalStorageEnumerator.cs]: @hez2010 I've noticed that the code to load thumbail inside `AddFileAsync` was never called (the function is always ever called with `suppressThumbnailLoading=true`) and removed it. Please let me know if you agree with the changes!

Comment (hez2010) [reply Files/Filesystem/StorageEnumerators/UniversalStorageEnumerator.cs]: BitmapImage need to be constructed in main thread, so I yield first to switch to main thread.

Comment (hez2010) [reply Files/Filesystem/StorageEnumerators/UniversalStorageEnumerator.cs]: LGTM

**Review by gave92 — APPROVED** [review]

**Comment by hez2010** [comment]
@yair100 Just tested with Release build. I think it's ready for merge.

**Pull request merged**